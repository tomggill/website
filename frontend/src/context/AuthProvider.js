import React, { createContext, useMemo, useState } from 'react';

const AuthContext = createContext({});

export function AuthProvider({ children }) {
  const [auth, setAuth] = useState({});
  const [persist, setPersist] = useState(JSON.parse(localStorage.getItem('persist')) || false);

  return (
    <AuthContext.Provider value={useMemo(() => ({
      auth, setAuth, persist, setPersist,
    }), [auth, setAuth, persist, setPersist])}
    >
      {children}
    </AuthContext.Provider>
  );
}

export default AuthContext;
