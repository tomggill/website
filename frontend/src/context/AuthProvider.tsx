import React, {
  ReactNode, createContext, useMemo, useState,
} from 'react';

interface AuthDetails {
  username: string;
  accessToken: string;
}

type SetAuthFunction = (auth: AuthDetails | {} | ((prev: AuthDetails) => AuthDetails)) => void;

export type AuthContextType = {
  auth: AuthDetails;
  setAuth: SetAuthFunction;
  persist: boolean;
  setPersist: (updatePersist: (prev: boolean) => boolean) => void;
};

interface AuthProviderProps {
  children: ReactNode;
}

const AuthContext = createContext<AuthContextType | {}>({});
// const AuthContext = createContext({});

export function AuthProvider({ children }: AuthProviderProps) {
  const [auth, setAuth] = useState({});
  const [persist, setPersist] = useState(JSON.parse(localStorage.getItem('persist')!) || false);

  return (
    <AuthContext.Provider value={useMemo(() => ({
      auth, setAuth, persist, setPersist,
    }), [auth, setAuth, persist, setPersist])}
    >
      {children}
    </AuthContext.Provider>
  );
}

export default AuthContext;
