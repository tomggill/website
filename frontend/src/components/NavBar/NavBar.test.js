import './matchMediaMock';
import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './NavBar';
import { AuthProvider } from '../../context/AuthProvider';
import '@testing-library/jest-dom';

let mockAuth = {};

jest.mock('../../hooks/useAuth', () => ({
  __esModule: true,
  default: () => ({ auth: mockAuth }),
}));

afterEach(() => {
  cleanup();
  mockAuth = {};
});

test('Renders NavBar component correctly without an access token present', () => {
  render(
    <Router>
      <AuthProvider>
        <Navbar />
      </AuthProvider>
    </Router>,
  );
  const navBarSignUpElement = screen.getByTestId('navLinkSignUp');
  const navBarSignInElement = screen.getByTestId('navLinkSignIn');
  const navBarSignOutElement = screen.queryByTestId('navLinkSignOut');
  expect(navBarSignUpElement).toBeInTheDocument();
  expect(navBarSignInElement).toBeInTheDocument();
  expect(navBarSignOutElement).toBeNull();
});

test('Renders NavBar component correctly with an access token present', () => {
  mockAuth = {
    username: 'testUsername',
    accessToken: 'testAccessToken',
  };
  render(
    <Router>
      <AuthProvider>
        <Navbar />
      </AuthProvider>
    </Router>,
  );
  const navBarSignUpElement = screen.queryByTestId('navLinkSignUp');
  const navBarSignInElement = screen.queryByTestId('navLinkSignIn');
  const navBarSignOutElement = screen.getByTestId('navLinkSignOut');
  expect(navBarSignUpElement).toBeNull();
  expect(navBarSignInElement).toBeNull();
  expect(navBarSignOutElement).toBeInTheDocument();
});

test('NavBar snapshot test', () => {
  const component = renderer.create(
    <Router>
      <AuthProvider>
        <Navbar />
      </AuthProvider>
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
