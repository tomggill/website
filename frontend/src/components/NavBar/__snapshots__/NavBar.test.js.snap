// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot test 1`] = `
<nav
  className="sc-beyTiQ jSXkOa"
>
  <svg
    className="sc-dmyDmy kvijpn"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
    />
  </svg>
  <div
    className="sc-hLQTFJ jFuntT"
  >
    <a
      className="sc-guDMob ncYGY"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="sc-guDMob ncYGY"
      href="/annual"
      onClick={[Function]}
    >
      Annual Report
    </a>
    <a
      className="sc-guDMob ncYGY"
      href="/team"
      onClick={[Function]}
    >
      Teams
    </a>
    <a
      className="sc-guDMob ncYGY"
      href="/react-helper"
      onClick={[Function]}
    >
      React Helper
    </a>
  </div>
  <nav
    className="sc-eDLKEg hkzCqS"
  >
    <button
      className="sc-iBdoyZ jVgAuz"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.993 12c0 2.761 2.246 5.007 5.007 5.007s5.007-2.246 5.007-5.007S14.761 6.993 12 6.993 6.993 9.239 6.993 12zM12 8.993c1.658 0 3.007 1.349 3.007 3.007S13.658 15.007 12 15.007 8.993 13.658 8.993 12 10.342 8.993 12 8.993zM10.998 19h2v3h-2zm0-17h2v3h-2zm-9 9h3v2h-3zm17 0h3v2h-3zM4.219 18.363l2.12-2.122 1.415 1.414-2.12 2.122zM16.24 6.344l2.122-2.122 1.414 1.414-2.122 2.122zM6.342 7.759 4.22 5.637l1.415-1.414 2.12 2.122zm13.434 10.605-1.414 1.414-2.122-2.122 1.414-1.414z"
        />
      </svg>
    </button>
    <a
      className="sc-jTQDJr hRBcQE"
      data-testid="navLinkSignUp"
      href="/register"
      onClick={[Function]}
    >
      Sign Up
    </a>
    <a
      className="sc-jTQDJr hRBcQE"
      data-testid="navLinkSignIn"
      href="/login"
      onClick={[Function]}
    >
      Sign In
    </a>
  </nav>
</nav>
`;
